docs/blockbuilder.md

```diff
@@ -0,0 +1,251 @@
+# Block Builder
+
+- [Block Builder](#block-builder)
+  - [Included Blocks](#included-blocks)
+      - [Heading Block `<h1>, <h2>, <h3>, <h4>, <h5>, <h6>`](#heading-block-h1-h2-h3-h4-h5-h6)
+      - [Table Block `<table>`](#table-block-table)
+      - [Iframe Block `<iframe>`](#iframe-block-iframe)
+      - [Form Block `<form>`](#form-block-form)
+      - [Image Block `<img />` `TODO not yet complete, likely to come from shortcode parsing`](#image-block-img--todo-not-yet-complete-likely-to-come-from-shortcode-parsing)
+      - [Blockquote Block `<blockquote>`](#blockquote-block-blockquote)
+  - [Included Fallback/Catch-all Block](#included-fallbackcatch-all-block)
+  - [Configuration](#configuration)
+
+The block builder takes the page body content in as a string of HTML.
+The filters listed below are then used to parse the HTML into a sequence of StreamField blocks.
+
+The parsing process uses Beautiful Soup to analyze each top level HTML tag in the order they appear in the HTML body content. If a match is found in the `WAGTAIL_WORDPRESS_IMPORTER_CONVERT_HTML_TAGS_TO_BLOCKS` configuration a single block is created for the HTML tag.
+
+---
+
+## Included Blocks
+
+#### Heading Block `<h1>, <h2>, <h3>, <h4>, <h5>, <h6>` 
+
+Builder:
+
+```python
+def build_heading_block(tag):
+    block_dict = {
+        "type": "heading",
+        "value": {"importance": tag.name, "text": tag.text},
+    }
+    return block_dict
+```
+
+Wagtail Block:
+
+```python
+class HeadingBlock(blocks.StructBlock):
+    text = blocks.CharBlock(classname="title")
+    importance = blocks.ChoiceBlock(
+        choices=(
+            ("h1", "H1"),
+            ("h2", "H2"),
+            ("h3", "H3"),
+            ("h4", "H4"),
+            ("h5", "H5"),
+            ("h6", "H6"),
+        ),
+        default="h1",
+    )
+
+    class Meta:
+        icon = "title"
+        template = "wagtail_wordpress_import/heading_block.html"
+```
+
+#### Table Block `<table>`
+Filter: 
```

Comment:
**Question**: Does the mkdocs markdown parser cope with missing blank lines after headings? And before triple-backtick code blocks?

Reply:
Yes you are right. Checked it with https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint and fixed